filelog_unix.go

```diff
@@ -2,4 +2,69 @@
 
 package log4go
 
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
 const FILELOG_ARCHIVE_REGEX = `\.[0-9]{4}-[0-9]{2}-[0-9]{2}(\.[0-9]{4})?(\.gz)?$`
+
+func (w *FileLogWriter) compressFile(filename string) {
+	basename := filepath.Base(filename)
+	dirname := filepath.Dir(filename)
+
+	plainFile, err := os.Open(filename)
+	if err != nil {
+		fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't open logfile to begin compression: %s\n", w.filename, err)
+		return
+	}
+
+	compressedFilename := filepath.Join(dirname, basename+".gz")
+	compressedFile, err := os.Create(compressedFilename + ".inprogress")
+	if err != nil {
+		fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't open new compressed file: %s\n", w.filename, err)
+		return
+	}
+	compressedFileWriter := gzip.NewWriter(compressedFile)
+
+	// Read plain file, write to compressed file
+	_, err = io.Copy(compressedFileWriter, plainFile)
+	if err != nil {
+		fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't write compressed file: %s\n", w.filename, err)
```

Comment:
Should we attempt to clean up the gzipped file if we fail to compress the whole contents?

edit: We definitely need to close it.